ecl/hthor/hthor.cpp

```diff
@@ -4929,7 +4953,7 @@ bool CHThorSelfJoinActivity::fillGroup()
     failingOuterAtmost = false;
     OwnedConstRoxieRow next;
     unsigned groupCount = 0;
-    next.setown(input->nextInGroup());
+    next.setown(groupedInput->nextInGroup());
```

Comment:
I probably meant dualCacheInput. Interesting that the regression suite did not pick it up if it is wrong though... I'll stare at it some more to be sure.